addons/hr_attendance/controllers/main.py

```diff
@@ -9,7 +9,31 @@
 
 import datetime
 
-class HrAttendance(http.Controller):
+from odoo.addons.mail.controllers.webclient import WebclientController
+from odoo.addons.mail.tools.discuss import Store
+
+
+class HrAttendance(WebclientController):
+    @classmethod
+    def _process_request_for_internal_user(self, store: Store, name, params):
+        super()._process_request_for_internal_user(store, name, params)
+        if name == "/hr_attendance/user_data":
+            employee = request.env.user.employee_id
+            store.add_global_values(
+                self_employee=Store.One(employee, self._get_user_attendance_data(employee))
```

Comment:
Note: I wish I could use actual fields instead of `_get_user_attendance_data` but it's used inside `_get_employee_info_response` of kiosk too and I don't want to touch that code in this PR.